src/platform/nm-platform.c

```diff
@@ -6369,17 +6369,40 @@ const char *
 nm_platform_qdisc_to_string (const NMPlatformQdisc *qdisc, char *buf, gsize len)
 {
 	char str_dev[TO_STRING_DEV_BUF_SIZE];
+	char opt_buf[300] = { 0, };
+	char *p = opt_buf;
+	gsize l = sizeof (opt_buf);
 
 	if (!nm_utils_to_string_buffer_init_null (qdisc, &buf, &len))
 		return buf;
 
-	g_snprintf (buf, len, "%s%s family %d handle %x parent %x info %x",
+	if (nm_streq (qdisc->kind, "fq_codel")) {
+		if (qdisc->fq_codel.limit)
+			nm_utils_strbuf_append (&p, &l, " limit %d", qdisc->fq_codel.limit);
+		if (qdisc->fq_codel.flows)
+			nm_utils_strbuf_append (&p, &l, " flows %d", qdisc->fq_codel.flows);
+		if (qdisc->fq_codel.target)
+			nm_utils_strbuf_append (&p, &l, " target %d", qdisc->fq_codel.target);
+		if (qdisc->fq_codel.interval)
+			nm_utils_strbuf_append (&p, &l, " interval %d", qdisc->fq_codel.interval);
+		if (qdisc->fq_codel.quantum)
+			nm_utils_strbuf_append (&p, &l, " quantum %d", qdisc->fq_codel.quantum);
+		if (qdisc->fq_codel.ce_threshold != -1)
+			nm_utils_strbuf_append (&p, &l, " ce_threshold %d", qdisc->fq_codel.ce_threshold);
+		if (qdisc->fq_codel.memory != -1)
+			nm_utils_strbuf_append (&p, &l, " memory %d", qdisc->fq_codel.memory);
+		if (qdisc->fq_codel.ecn)
+			nm_utils_strbuf_append (&p, &l, " ecn");
+	}
+
+	g_snprintf (buf, len, "%s%s family %d handle %x parent %x info %x%s",
 	            qdisc->kind,
 	            _to_string_dev (NULL, qdisc->ifindex, str_dev, sizeof (str_dev)),
 	            qdisc->addr_family,
 	            qdisc->handle,
 	            qdisc->parent,
-	            qdisc->info);
+	            qdisc->info,
+	            opt_buf);
```

Comment:
`opt_buf` is a buffer that's appended at the end. So, replace `g_snprintf()` by `nm_utils_strbuf_append()`, and then don't use `opt_buf`, but let it append to the correct buffer right away.